server-core/engine/src/main/java/io/infectnet/server/engine/content/system/hijack/HijackSystem.java

```diff
@@ -0,0 +1,32 @@
+package io.infectnet.server.engine.content.system.hijack;
+
+
+import io.infectnet.server.engine.content.system.health.HealthModificationRequest;
+import io.infectnet.server.engine.core.entity.wrapper.Action;
+import io.infectnet.server.engine.core.script.Request;
+import io.infectnet.server.engine.core.system.ActionOnlyProcessor;
+import io.infectnet.server.engine.core.util.ListenableQueue;
+
+public class HijackSystem extends ActionOnlyProcessor {
+
+  private static final int MODIFICATION_NUMBER = 3;
+
+  private final ListenableQueue<Request> requestQueue;
+
+  public HijackSystem(
+      ListenableQueue<Request> requestQueue) {
+    this.requestQueue = requestQueue;
+  }
+
+  @Override
+  public void registerActionListeners(ListenableQueue<Action> actionQueue) {
+    actionQueue.addListener(HijackAction.class, this::consumeHijackAction);
+  }
+
+  private void consumeHijackAction(Action action) {
+    HijackAction hijackAction = (HijackAction) action;
```

Comment:
Maybe we should check if the target of the action is a worker, a fighter or a building. Resources and specials should not be hijacked if you agree.